src/sensors/Particles.js

```diff
@@ -61,43 +61,61 @@ ROS3D.Particles = function(options) {
     '}'
     ].join('\n');
 
-    this.geom = new THREE.Geometry();
-    for(var i=0;i<this.max_pts;i++){
-        this.geom.vertices.push(new THREE.Vector3( ));
-    }
-
     var customUniforms =
     {
-        texture:   { type: 't', value: THREE.ImageUtils.loadTexture( texture ) },
+		texture:   { type: 't', value: new THREE.TextureLoader().load( texture ) }
+        //texture:   { type: 't', value: THREE.ImageUtils.loadTexture( texture ) },
     };
-
-    this.attribs =
+	
+	
+    /*this.attribs =
     {
         customColor:   { type: 'c', value: [] },
         alpha:         { type: 'f', value: [] }
-    };
+    };*/
 
     this.shaderMaterial = new THREE.ShaderMaterial(
     {
         uniforms:          customUniforms,
-        attributes:        this.attribs,
         vertexShader:      this.vertex_shader,
         fragmentShader:    this.fragment_shader,
         transparent: true,
     });
+    
+    this.geom = new THREE.BufferGeometry();
+    
+    var positions = [];
+    var customColor = [];
+    var alpha = [];
+
+    for(var i = 0; i < this.max_pts; i++){
```

Comment:
Could this perhaps be done with something like
positions = new Array(3 * this.max_pts).fill(0); 

Likely to be significantly more efficient with the browsers array fill. 
It's reasonably well supported. I'm not sure how far back this project expects browser support, but it's been around for a while

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill